homeassistant/components/amazon_devices/config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Config flow for Amazon Devices integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+import aioamazondevices
+from aioamazondevices import AmazonEchoApi
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_CODE, CONF_COUNTRY, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+
+from .const import _LOGGER, CONF_LOGIN_DATA, DEFAULT_COUNTRY, DOMAIN
+
+
+def user_form_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Return user form schema."""
+    user_input = user_input or {}
+    return vol.Schema(
+        {
+            vol.Required(CONF_COUNTRY, default=DEFAULT_COUNTRY): cv.country,
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Required(CONF_CODE): cv.positive_int,
+        }
+    )
+
+
+STEP_REAUTH_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CODE): cv.positive_int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
Because we raise exceptions, inlining it might even save some lines of code

Reply:
same code is used twice